doc/user/content/sql/functions/now_and_mz_now.md

```diff
@@ -14,86 +14,120 @@ By contrast, `mz_now()` returns the logical time at which the query was executed
+  An ad-hoc `SELECT` query with `now()` and `mz_now()` can be useful if you need to understand how up to date the data returned by a query is.
+  The data returned by the query reflects the results as of the logical time returned by a call to `mz_now()` in that query.
 
-* **Temporal filters**
 
-  Various windowing idioms involve using `mz_now()` in a [temporal filter](/sql/patterns/temporal-filters).
+{{< warning >}}
+Queries that contain `now()` or `mz_now()` in the `SELECT` clause cannot be materialized.
+In other words, you cannot create an index or a materialized view on a query with `now()` or `mz_now()` in the `SELECT` clause.
+This is because `now()` and `mz_now()` change every millisecond, so if this materialization _were_ allowed, every record in the collection would be updated every millisecond, which would be resource prohibitive.
+{{< /warning >}}
 
-## Example
+## Examples
 
-### Temporal filter using mz_now()
+### Consider Recent Data with a Temporal Filter
 
 <!-- This example also appears in temporal-filters -->
-For this example, you'll need to create a sample data source and create an indexed view from it for later reference.
+It is common for real-time applications to be concerned with only a recent period of time.
+In this case, we will filter a table to only include only records from the last 30 seconds.
 
 ```sql
---Create a table of timestamped events.
+-- Create a table of timestamped events.
 CREATE TABLE events (
-    content text,
-    insert_t timestamp,
-    delete_t timestamp
+    content TEXT,
+    event_ts TIMESTAMP
 );
 
---Create an indexed view of events valid at a given logical time.
-CREATE VIEW valid AS
-SELECT content, insert_t, delete_t
+-- Create a view of events from the last 30 seconds.
+CREATE VIEW last_30_sec AS
+SELECT event_ts, content
 FROM events
-WHERE mz_now() >= insert_t
-  AND mz_now() < delete_t;
-CREATE DEFAULT INDEX ON valid;
+WHERE mz_now() <= event_ts + INTERVAL '30s';
+```
+
+Next, subscribe to the results of the view.
+
+```sql
+COPY (SUBSCRIBE (SELECT event_ts, content FROM last_30_sec)) TO STDOUT;
 ```
 
-Next, you'll populate the table with timestamp data.
+In a separate session, insert a record.
 
 ```sql
 INSERT INTO events VALUES (
     'hello',
-    now(),
-    now() + '100s'
+    now()
+);
+```
+
+Back in the first session, watch the record expire after 30 seconds. Press `Ctrl+C` to quit the `SUBSCRIBE` when you are ready.
+
+```nofmt
+1686868190714   1       2023-06-15 22:29:50.711 hello
+1686868220712   -1      2023-06-15 22:29:50.711 hello
+```
+
+You can materialize the `last_30_sec` view by creating an index on it (results stored in memory) or by recreating it as a `MATERIALIZED VIEW` (results persisted to storage). When you do so, Materialize will keep the results up to date with records expiring automatically according to the temporal filter.
+
+### Ad-hoc queries with now() and mz_now()
```

Comment:
```suggestion
### Query timestamp introspection
```

Reply:
Suggestion, but these should match the two use cases you set above, so it's easier to make the link between explanation->examples.